### DISCUSSION TITLE
Multiple filters in onPostgresChange subscriber

### DISCUSSION THREAD

**Discussion by sbmohan1** [supabase/supabase#25906] [Questions] [answered]
In my SwiftUI app, I want to listen to inserts in my table called user_interactions:

```
 subscription = await channel.onPostgresChange(
                    InsertAction.self,
                    schema: "public",
                    table: "user_interactions",
                    filter: "profile_id=eq.\(userId)"
                ) { insert in
                    print("Insert detected: User \(insert.record["user_id"] ?? "unknown") liked \(userId)")
                }
                await channel.subscribe()
```
 The code above works as expected, but I would ideally like to add another filter on a column called action_type. So I tried to add the following line as the filter:
 `filter: "profile_id=eq.\(userId) & action_type=eq.Like"`
 
 However multiple filters don't seem to be working. Is that a limitation of supabase or is my syntax incorrect?

**Comment by GaryAustin1** [comment, marked as answer]
Realtime only allows 1 filter out of the 6 or 7 types in the docs.
